linkerd/tls/src/client.rs

```diff
@@ -38,11 +46,16 @@ pub enum NoClientTls {
     NotProvidedByServiceDiscovery,
 }
 
-/// Indicates whether the target server endpoint has a known TLS identity.
-pub type ConditionalClientTls = Conditional<ServerId, NoClientTls>;
+/// A stack paramater that indicates whether the target server endpoint has a
+/// known TLS identity.
+pub type ConditionalClientTls = Conditional<ClientTls, NoClientTls>;
 
 pub type Config = Arc<rustls::ClientConfig>;
 
+/// A stack param that configures the available ALPN protocols.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
Does it make sense for this to have a manual `Debug` impl instead, so that each protocol can be formatted using `NegotiatedProtocolRef` or something, rather than as a pile of bytes?